core/src/transaction_builder.rs

```diff
@@ -72,19 +87,38 @@ pub fn create_taproot_address(
     (taproot_address, tree_info)
 }
 
-/// Generates a deposit address for the user. N-of-N or user takes after
-/// timelock script can be used to spend the funds.
-#[tracing::instrument(ret(level = tracing::Level::TRACE))]
+/// Generates a deposit address for the user. Funds can be spend by N-of-N or
+/// user can take after specified time.
+///
+/// # Parameters
+///
+/// - `nofn_xonly_public_key`: N-of-N x-only public key of the depositor
+/// - `recovery_taproot_address`: User's x-only public key that can be used to
+///   take funds after some time
+/// - `user_evm_address`: User's EVM address.
+/// - `amount`: Amount to deposit (in sats)
+/// - `network`: Bitcoin network to work on
+/// - `user_takes_after`: User can take the funds back, after this amounts of
+///   blocks have passed
+///
+/// # Returns
+///
+/// - [`Address`]: Deposit taproot Bitcoin address
+/// - [`TaprootSpendInfo`]: Deposit address's taproot spending information
+///
+/// # Panics
+///
+/// Panics if given parameters are malformed.
 pub fn generate_deposit_address(
-    nofn_xonly_pk: &XOnlyPublicKey,
+    nofn_xonly_public_key: XOnlyPublicKey,
```

Comment:
let use nofn_xonly_pk everywhere for consistency

Reply:
c81b37d2355697129e7c8e7c9694ee4e1e5e6a46